pyproject.toml

```diff
@@ -1,3 +1,81 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The Foundry Visionmongers Ltd
+
+# The metadata here should be kept in sync with the main OpenAssetIO
+# repository (Python versions, platforms, etc): https://github.com/OpenAssetIO/OpenAssetIO
+
+[build-system]
+requires = [
+    "setuptools>=65.5.0"
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "openassetio-traitgen"
+version = "1.0.0a3"
+requires-python = ">=3.7"
+
+authors = [
+    { name = "Contributors to the OpenAssetIO project", email = "openassetio-discussion@lists.aswf.io" }
+]
+description = """\
+    Generate OpenAssetIO Trait and Specification classes from a simple YAML description.\
+    """
+keywords = ["openassetio", "codegen", "trait"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Build Tools",
+    "License :: OSI Approved :: Apache Software License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```

Comment:
[Question] Is this still correct for a source only build? Is there any reason to exclude 3.8 anymore? I guess it isn't tested...

Reply:
Good point.  It's gonna be easier to maintain if we just remove all the specific version lines, leaving just `Python :: 3`, will do that.

Reply:
Note that we have `requires-python = ">=3.7"` to signal the minimum supported version, and this shows up in PyPI